### PR TITLE
Fixes a crash on Android 6 and 7 devices on a DApps panel

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
Resolves https://github.com/brave/brave-browser/issues/23215

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java [hunk 1]
```diff
@@ -119,41 +120,53 @@ public void onDismiss() {
...
+            if (menu.getMenu() instanceof MenuBuilder) {
+                ((MenuBuilder) menu.getMenu()).setOptionalIconsVisible(true);
+            }
+            menu.show();
+        } else {
+            android.widget.PopupMenu menu = new android.widget.PopupMenu(
+                    mAnchorViewHost.getContext(), (View) mAnchorViewHost);
+            menu.getMenuInflater().inflate(R.menu.menu_dapps_panel, menu.getMenu());
+            menu.setOnMenuItemClickListener(item -> { return handleMenuItemClick(item); });
+            menu.show();
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java:139]: not sure but might be possible to use internal [Menubuilder](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/com/android/internal/view/menu/MenuBuilder.java;drc=349d6efeed02e20c23093f73b3d1ea913ec43632;l=53) here to show the icon. also not sure if it shows the icon without the menubuilder?

File: android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java [hunk 2]
```diff
@@ -119,41 +120,53 @@ public void onDismiss() {
...
     }
 
-    public void showPopupMenu() {
-        PopupMenu menu = new PopupMenu(mOptionsImage.getContext(), (View) mOptionsImage);
-        menu.getMenuInflater().inflate(R.menu.menu_dapps_panel, menu.getMenu());
-        menu.setOnMenuItemClickListener(item -> {
-            if (item.getItemId() == R.id.action_lock_wallet) {
-                mBraveWalletPanelServices.getKeyringService().lock();
+    private void showPopupMenu() {
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.N_MR1) {
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java:124]: nit: would be great to add a comment here about the crash and what exactly fixes it, as it seems like a typical android edge case (bug in the platform lib).

**Review by Pavneet-Sing — APPROVED** [review]

File: android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java (hunk 1 above)
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java:139]: yeah, not sure, anyways, I fixed the crash, we can always polish the UI more if we need to, we have only a few % of users on Android 6 and 7, I don't think it's worth spending time on polishing that more. The main functionality is working.

File: android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java (hunk 2 above)
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java:124]: it's a good point, let me add it.

**Review by qamarngr — APPROVED** [review]
++

**Pull request merged**